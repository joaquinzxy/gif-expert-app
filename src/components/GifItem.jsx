import PropTypes from 'prop-types';


export const GifItem = ({ url }) => {

    return (
    <a className="card" href={url}>
        <img src={url}/>
    </a>
  )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}