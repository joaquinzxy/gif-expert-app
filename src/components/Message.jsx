export const Message = ({ message }) => {
  return (
    <div className='message-container'>
        <h3>{message}</h3>
    </div>
  )
}
