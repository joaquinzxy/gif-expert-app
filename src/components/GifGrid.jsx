import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";
import { PropTypes } from "prop-types";
import { useState } from "react";

export const GifGrid = ({ category, onDeleteCategory }) => {

  const gifsPerPage = 4;

  const { images, isLoading } = useFetchGifs(category);

  const [index, setIndex] = useState(0);

  const [paginationToggle, setPaginationToggle] = useState([true, false])

  const nextHandler = () => {
    setIndex(index + gifsPerPage);
    index < 36 ? setPaginationToggle([false, false]) : setPaginationToggle([false, true])
  }

  const previousHandler = () => {
    setIndex(index - gifsPerPage);
    index > gifsPerPage ? setPaginationToggle([false, false]) : setPaginationToggle([true, false])
  }

  return (
    <>
      <div className="gif-grid">
        <div className="gallery-header">
          <h4>▶️ {category}</h4>
          <div className="pagination">
            <button onClick={previousHandler} disabled={paginationToggle[0]}>←</button>
            <button onClick={nextHandler} disabled={paginationToggle[1]}>→</button>
            <button onClick={() => onDeleteCategory(category)} className={'close-btn'}> ✕ </button>
          </div>
        </div>
        {
          isLoading && (<h2>Cargando...</h2>)
        }
        <div className="card-grid">
          {
            images.slice(index, index + 4).map((image) => (
              <GifItem
                key={image.id}
                {...image} />
            ))
          }
        </div>
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}
