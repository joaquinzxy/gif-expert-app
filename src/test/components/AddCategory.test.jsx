import { fireEvent, render, screen } from "@testing-library/react";
import { AddCategory } from "../../components";

describe(' test addCategory', () => {

    const inputValue = 'Saitama';

    test('should match snapshot', () => {

        const { content } = render(<AddCategory onNewCategory={() => { }} />)
        expect(content).toMatchSnapshot();

    })

    test('should change value on input', () => {

        render(<AddCategory onNewCategory={() => { }} />)
        const input = screen.getByRole("textbox");

        fireEvent.input(input, { target: { value: inputValue } })

        expect(input.value).toBe(inputValue);

    })

    test('should call onNewCategory if > 1', () => {

        const onNewCategory = jest.fn();

        render(<AddCategory onNewCategory={onNewCategory} />)

        const input = screen.getByRole("textbox");
        const form = screen.getByRole('form')

        fireEvent.input(input, { target: { value: inputValue } })
        fireEvent.submit(form)

        expect(input.value).toBe('');
        expect(onNewCategory).toHaveBeenCalledTimes(1);
        expect(onNewCategory).toHaveBeenCalledWith(inputValue.toUpperCase())


    })

    test('should not called onNewCategory if input empty', () => {

        const onNewCategory = jest.fn();

        render(<AddCategory onNewCategory={onNewCategory} />);

        const input = screen.getByRole("textbox");
        const form = screen.getByRole('form')

        fireEvent.submit(form)
        expect(onNewCategory).toHaveBeenCalledTimes(0);

    })
})
