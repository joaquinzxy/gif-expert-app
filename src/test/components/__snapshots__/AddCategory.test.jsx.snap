// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` test addCategory should match snapshot 1`] = `undefined`;
